tests/test_h2_client.c

```diff
@@ -301,6 +301,54 @@ TEST_CASE(h2_client_stream_complete) {
     return s_tester_clean_up();
 }
 
+/* Test that h2 stream can take a h1 request massega and transfrom it to h2 style to send it. */
+TEST_CASE(h2_client_stream_with_h1_request_message) {
```

Comment:
Do we need a test for if the headers were explicitly set on the request?

Reply:
The other tests of the stream set pseudo headers explicitly on the request. Maybe we just let those to test that situation.